gateway/src/apicast/balancer.lua

```diff
@@ -1,18 +1,53 @@
 local round_robin = require 'resty.balancer.round_robin'
+local resty_url = require 'resty.url'
+local empty = {}
 
 local _M = { default_balancer = round_robin.new() }
 
+local function get_default_port(upstream_url)
+  local url = resty_url.split(upstream_url) or empty
+  local scheme = url[1] or 'http'
+  return resty_url.default_port(scheme)
```

Comment:
Just to clarify, I think this only works with http and https: https://github.com/3scale/lua-resty-url#default_port

Reply:
Right. Do you think it needs to be documented somewhere in this PR?

Reply:
Well spotted. I think it does not need to be documented (because we don't really support other protocols), but handled with a proper error message.

Reply:
btw, `set_current_peer` will set port to `0` if not provided: https://github.com/openresty/lua-resty-core/blob/a39b8d68b3b1c93f22259f95925c9dc854d69c1c/lib/ngx/balancer.lua#L113

Reply:
I guess then we can just leave it to that and just have a test to verify what it does when no port is provided.

Reply:
@mikz I've now fixed it to return an error, is that fine? can still write a test for that

Reply:
If the balancer has a default and would return an error then there is no need to handle this case by us. And to verify that it works it is necessary to have a test. IMO it is better to have a test to see how it works than code to maintain.

Reply:
@mayorova fail how? throw an exception? then it should be handled and we can test it is being handled by asserting an entry in the error log and some status.

Reply:
@mikz 1.  will throw an exception...
My point is – I don't see a way to test how *balancer* will act if the incorrect scheme is passed, because the execution will not even reach the balancer point, and will fail before that.
Or should I add a failing test for the upstream policy instead? (I would consider it as a different PR - handle exception in upstream policy and add test)

Reply:
@mayorova does https://github.com/3scale/apicast/blob/dab3c183144fa4dc74e765e147ac56297cdcfa05/t/apicast-upstream-balancer.t help?